basic/src/test/java/org/jboss/hal/testsuite/test/configuration/io/BufferPoolTestCase.java

```diff
@@ -0,0 +1,178 @@
+        final int value = 10;
+        ModelNodeResult modelNodeResult = operations.readAttribute(BUFFER_POOL_ADDRESS, BUFFERS_PER_SLICE);
+        modelNodeResult.assertSuccess();
+        try {
+            new ConfigChecker.Builder(client, BUFFER_POOL_ADDRESS)
+                    .configFragment(page.getConfigFragment())
+                    .editAndSave(ConfigChecker.InputType.TEXT, BUFFERS_PER_SLICE, String.valueOf(value))
+                    .verifyFormSaved()
+                    .verifyAttribute(BUFFERS_PER_SLICE, value);
+        } finally {
+            operations.writeAttribute(BUFFER_POOL_ADDRESS, BUFFERS_PER_SLICE, modelNodeResult.value());
+        }
+    }
+
+    @Test
+    public void editBuffersPerSliceInvalid() throws Exception {
+        final int value = -1;
+        ModelNodeResult modelNodeResult = operations.readAttribute(BUFFER_POOL_ADDRESS, BUFFERS_PER_SLICE);
+        modelNodeResult.assertSuccess();
+        try {
+            new ConfigChecker.Builder(client, BUFFER_POOL_ADDRESS)
+                    .configFragment(page.getConfigFragment())
+                    .editAndSave(ConfigChecker.InputType.TEXT, BUFFERS_PER_SLICE, String.valueOf(value))
+                    .verifyFormNotSaved();
+        } finally {
+            operations.writeAttribute(BUFFER_POOL_ADDRESS, BUFFERS_PER_SLICE, modelNodeResult.value());
+        }
+    }
+
+    @Test
+    public void editBufferSize() throws Exception {
+        final int value = 1024;
+        ModelNodeResult modelNodeResult = operations.readAttribute(BUFFER_POOL_ADDRESS, BUFFER_SIZE);
+        modelNodeResult.assertSuccess();
+        try {
+            new ConfigChecker.Builder(client, BUFFER_POOL_ADDRESS)
+                    .configFragment(page.getConfigFragment())
+                    .editAndSave(ConfigChecker.InputType.TEXT, BUFFER_SIZE, String.valueOf(value))
+                    .verifyFormSaved()
+                    .verifyAttribute(BUFFER_SIZE, value);
+        } finally {
+            operations.writeAttribute(BUFFER_POOL_ADDRESS, BUFFER_SIZE, modelNodeResult.value());
+        }
+    }
+
+    @Test
+    public void editBufferSizeInvalid() throws Exception {
+        final int value = -1;
+        ModelNodeResult originalModelNodeResult = operations.readAttribute(BUFFER_POOL_ADDRESS, BUFFER_SIZE);
+        originalModelNodeResult.assertSuccess();
+        try {
+            new ConfigChecker.Builder(client, BUFFER_POOL_ADDRESS)
+                    .configFragment(page.getConfigFragment())
+                    .editAndSave(ConfigChecker.InputType.TEXT, BUFFER_SIZE, String.valueOf(value))
+                    .verifyFormNotSaved();
+        } finally {
+            operations.writeAttribute(BUFFER_POOL_ADDRESS, BUFFER_SIZE, originalModelNodeResult.value());
+        }
+    }
+
+    @Test
+    public void toggleDirectBuffers() throws Exception {
+        ModelNodeResult modelNodeResult = operations.readAttribute(BUFFER_POOL_ADDRESS, DIRECT_BUFFERS);
```

Comment:
Please rename to something like `originalDirectBuffersResult` or at least `originalResult`.